pkg/scheduler/plugins/task-topology/topology.go

```diff
@@ -247,10 +247,16 @@ func affinityCheck(job *api.JobInfo, affinity [][]string) error {
 
 	var taskNumber = len(job.Tasks)
 	var taskRef = make(map[string]bool, taskNumber)
+	var jobNamePrefix = job.Name + "-"
 	for _, task := range job.Tasks {
-		tmpStrings := strings.Split(task.Name, "-")
-		if _, exist := taskRef[tmpStrings[len(tmpStrings)-2]]; !exist {
-			taskRef[tmpStrings[len(tmpStrings)-2]] = true
+		i := len(task.Name) - 1
+		for ; task.Name[i] != '-'; i-- {
```

Comment:
How about use `strings.LastIndex` directly to find the last `-` index?

Reply:
Thanks. I fixed it.